src/hooks/use-is-valid-wp-cli-inline.ts

```diff
@@ -0,0 +1,32 @@
+import { parse } from 'shell-quote';
+
+const PLACEHOLDER_CHAR_BEGIN = [ '<', '[', '{', '(' ];
+
+export function useIsValidWpCliInline( command: string ) {
+	const wpCliArgs = parse( command )
+		.map( ( arg ) => {
+			if ( typeof arg === 'string' || arg instanceof String ) {
+				return arg;
+			} else if ( 'op' in arg ) {
+				return arg.op;
+			} else {
+				return false;
+			}
+		} )
+		.filter( Boolean ) as string[];
+	const wpCommandCount = wpCliArgs.filter( ( arg ) => arg === 'wp' ).length;
+	const containsPath = wpCliArgs.some(
+		( arg ) => /path/i.test( arg[ 0 ] ) || /\//.test( arg[ 0 ] )
+	);
+	const containsPlaceholderArgs = wpCliArgs.some( ( arg ) =>
+		PLACEHOLDER_CHAR_BEGIN.includes( arg[ 0 ] )
+	);
+	const isValidWpCliCommand =
+		wpCliArgs.length > 0 &&
+		wpCliArgs[ 0 ] === 'wp' &&
+		wpCommandCount === 1 &&
+		! containsPath &&
+		! containsPlaceholderArgs;
+
+	return isValidWpCliCommand;
```

Comment:
Nitpick: we could directly return the value without creating a constant.

```suggestion
	return (
		wpCliArgs.length > 0 &&
		wpCliArgs[ 0 ] === 'wp' &&
		wpCommandCount === 1 &&
		! containsPath &&
		! containsPlaceholderArgs
	);
```

Reply:
Done! https://github.com/Automattic/studio/pull/286/commits/682b0fbf21e0699da2c12248be1a3ae33babef22